tokio-console/src/state/resources.rs

```diff
@@ -71,27 +76,50 @@ struct ResourceStats {
 
 impl Default for SortBy {
     fn default() -> Self {
-        Self::Rid
+        Self::Id
     }
 }
 
 impl SortBy {
     pub fn sort(&self, now: SystemTime, resources: &mut [ResourceRef]) {
         match self {
-            Self::Rid => {
+            Self::Id => {
                 resources.sort_unstable_by_key(|resource| resource.upgrade().map(|r| r.borrow().id))
             }
+            Self::ParentId => resources.sort_unstable_by_key(|resource| {
+                resource.upgrade().map(|r| r.borrow().parent_id.clone())
+            }),
             Self::Kind => resources.sort_unstable_by_key(|resource| {
                 resource.upgrade().map(|r| r.borrow().kind.clone())
             }),
+            Self::Total => resources
+                .sort_unstable_by_key(|resource| resource.upgrade().map(|r| r.borrow().total(now))),
+            Self::Target => resources.sort_unstable_by_key(|resource| {
+                resource.upgrade().map(|r| r.borrow().target.clone())
+            }),
             Self::ConcreteType => resources.sort_unstable_by_key(|resource| {
                 resource.upgrade().map(|r| r.borrow().concrete_type.clone())
             }),
-            Self::Target => resources.sort_unstable_by_key(|resource| {
-                resource.upgrade().map(|r| r.borrow().target.clone())
+            Self::Visibility => resources
+                .sort_unstable_by_key(|resource| resource.upgrade().map(|r| r.borrow().visibility)),
+            Self::Location => resources.sort_unstable_by_key(|resource| {
+                resource.upgrade().map(|r| r.borrow().location.clone())
+            }),
+            Self::Attributes => resources.sort_unstable_by_key(|resource| {
+                resource.upgrade().map(|r| {
+                    // FIXME - we are taking only the first attribute as sorting key here,
+                    // and we are sorting each attribute as a String.
+                    // Instead, attributes should probably be parsed and sorted according to their actual values,
+                    // not just as strings.
+                    r.borrow()
+                        .formatted_attributes()
+                        .first()
+                        .into_iter()
+                        .flat_map(|v| v.iter())
+                        .map(|s| <Cow<str> as std::borrow::Borrow<str>>::borrow(&s.content))
+                        .collect::<String>()
```

Comment:
i'm not sure why the `collect::<String>`  call is necessary here. can't we just call `.next()` on this iterator, returning the first element, since there should only be a single element in the iterator? that way, we don't have to allocate a big pile of `String`s every time we sort the list...
```suggestion
                        .next()
```

Reply:
The thing is that `formatted_attributes()` returns a slice of vectors: `&[Vec<Span>]` , each one of the `Vec<Span>` representing an attribute, that is a key-value pair, such as `duration=7001ms`. In this case the vector is basically composed of `["duration", "=", "7001", "ms"]`.

Calling `next` instead of `collect` would basically just return the key of the attribute, `duration` in the example, which maybe could be fine for the moment, and then we could open an issue to do a more meaningful sorting of attributes.

Do you think that would be fine?

Reply:
Let's just sort by the first attribute name for now. We will need to consider the best way to sort that column, since different resources have different attributes anyway.

Could you please open an issue for the improvement and then you can reference it in the FIXME (it's always nice to have an issue for each FIXME).

Reply:
Thanks, sorting is now done only on the attribute name. Opened issue #496 and referenced it in the fixme